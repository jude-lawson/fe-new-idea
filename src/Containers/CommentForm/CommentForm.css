.comment--form-container {
  border-bottom: 1px solid grey;
}

.comment-form {
  width: 60%;
  margin: auto;
  padding: 2rem 4rem;
  display: flex;
  flex-direction: column;
  border-radius: 20px;
}

.comment--form-header {
  text-align: center;
  color: var(--primary-dark-grey);
  margin-bottom: 2rem;
}

.comment--body-input {
  height: 15rem;
  font-size: 14px;
  line-height: 1.4;
  padding: 3px 5px;
}

.comment--form-btn {
  margin-top: 4rem;
  color: var(--primary-blue);
  background: var(--primary-white);
  width: 12rem;
  padding: 1rem;
  margin: 2rem auto;
  font-weight: 700;
  border-radius: 23px;
}

.hidden {
  display: none;
}

.not-hidden {
  display: block;
}

#blue-btn {
  background-image: linear-gradient(to right, #289dc5, #50b4be);
  color: var(--primary-white);
}
