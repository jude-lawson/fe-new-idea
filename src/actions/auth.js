export const userLoginAction = user => ({
  type: 'LOGIN',
  user
});