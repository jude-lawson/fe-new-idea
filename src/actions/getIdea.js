export const getIdeaAction = idea => ({
  type: 'GET_IDEA',
  idea
});