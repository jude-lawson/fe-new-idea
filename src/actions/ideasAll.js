export const getIdeas = ideas => ({
  type: 'GET_IDEAS',
  ideas
});