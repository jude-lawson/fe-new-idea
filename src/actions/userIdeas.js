export const retrieveMyIdeas = ideas => ({
  type: 'RETRIEVE_MY_IDEAS',
  ideas
});