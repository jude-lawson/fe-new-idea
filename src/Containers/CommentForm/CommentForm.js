import React, { Component } from 'react';
import  {connect } from 'react-redux';
import { addComment } from '../../actions/comment';
import PropTypes from 'prop-types';

export class CommentForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      body: ''
    };
  }
  
  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }
  
  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addComment({...this.state, id: Date.now()});
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <textarea
          className="comment--body-input"
          placeholder="Description"
          name="body"
          onChange={this.handleChange}
        />
        <button>SUBMIT</button>
      </form>
    );
  }
}

CommentForm.propTypes = {
  addComment: PropTypes.func
};

export const mapStateToProps = (state) => ({
  comments: state.comments
});

export const mapDispatchToProps = (dispatch) => ({
  addComment: (comment) => dispatch(addComment(comment))
});

export default connect(mapStateToProps, mapDispatchToProps)(CommentForm);