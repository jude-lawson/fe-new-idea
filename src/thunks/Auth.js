import { userLoginAction, userLogOutAction } from '../actions/auth';
import { githubOAuthLogin, logout } from '../firebase/firebase';

export const firebaseLogin = () => {
  return async dispatch => {
    const newUser = await githubOAuthLogin();
    dispatch(userLoginAction(newUser));
  };
};

export const firebaseLogout = () => {
  return async dispatch => {
    await logout();
    dispatch(userLogOutAction());
  };
};