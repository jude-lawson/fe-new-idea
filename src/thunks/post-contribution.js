import { addComment } from '../actions/comment';

export const postContribution = comment => {
  return async dispatch => {
    const contribution = await jest.fn();
    console.log(contribution);
    // const comment = {...comment, ...contribution }
    dispatch(addComment(comment));
  };
};