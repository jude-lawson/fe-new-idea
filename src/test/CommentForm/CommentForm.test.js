import { CommentForm, mapStateToProps, mapDispatchToProps } from '../../Containers/CommentForm/CommentForm';
import { postContribution } from '../../thunks/post-contribution'
import { shallow, mount } from 'enzyme';
import React from 'react';
import { addComment } from '../../actions/comment';

describe('CommentForm', () => {
  let wrapper;
  const mockFunc = jest.fn();
  beforeEach(() => wrapper = shallow(<CommentForm postContribution={mockFunc} />))

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    let spy;
    let wrapper;

    const mockFunc = jest.fn();

    beforeEach(() => {
      wrapper = mount(<CommentForm postContribution={mockFunc}/>);
      spy = jest.spyOn(wrapper.instance(), 'handleChange');
    });

    it('should update state of email', () => {
      const mockEvent = {
        target: {
          value: 'h',
          name: 'body'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      wrapper.find('textarea').first().simulate('change');

      expect(spy).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should create a props object with the correct keys', () => {
      const mockState = {
        comments: [
          {
            body: 'I do or do not like your Idea [this] much.',
            id: 11
          }
        ]
      }
      const expected = {
        comments: [
        {
          body: 'I do or do not like your Idea [this] much.',
          id: 11
        }
        ]
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);

    });
  });


  test('should call postContribution on Submit', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
     body: 'I heart your tech idea'
    })
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockFunc).toHaveBeenCalled();
  })


  describe('mapDispatchToProps', () => {
    test('should call dispatch when using postContribution from MDTP', () => {
      const mockContribution = { body: 'Hello' };
      const mockDispatch = jest.fn();
      const postContribution = jest.fn();
      const actionToDispatch = postContribution(mockContribution, 6);
      const mappedToProps = mapDispatchToProps(mockDispatch);
      mappedToProps.postContribution();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});