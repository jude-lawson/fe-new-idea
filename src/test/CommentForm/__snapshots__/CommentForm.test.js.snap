// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="comment--form-header"
>
          Comment on the Idea
</h1>,
        <textarea
          className="comment--body-input"
          name="body"
          onChange={[Function]}
          placeholder="Description"
/>,
        <button
          className="comment--form-btn"
>
          SUBMIT
</button>,
      ],
      "className": "comment-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Comment on the Idea",
          "className": "comment--form-header",
        },
        "ref": null,
        "rendered": "Comment on the Idea",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "comment--body-input",
          "name": "body",
          "onChange": [Function],
          "placeholder": "Description",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SUBMIT",
          "className": "comment--form-btn",
        },
        "ref": null,
        "rendered": "SUBMIT",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="comment--form-header"
>
            Comment on the Idea
</h1>,
          <textarea
            className="comment--body-input"
            name="body"
            onChange={[Function]}
            placeholder="Description"
/>,
          <button
            className="comment--form-btn"
>
            SUBMIT
</button>,
        ],
        "className": "comment-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Comment on the Idea",
            "className": "comment--form-header",
          },
          "ref": null,
          "rendered": "Comment on the Idea",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "comment--body-input",
            "name": "body",
            "onChange": [Function],
            "placeholder": "Description",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SUBMIT",
            "className": "comment--form-btn",
          },
          "ref": null,
          "rendered": "SUBMIT",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
